lib/Conversion/TorchToArith/TorchToArith.cpp

```diff
@@ -189,8 +189,17 @@ class ConvertTorchTensorLiteralOp
               RankedTensorType::get(type.getShape(), builtinTensorElemTy);
           AsmResourceBlob *blob = elements.getRawHandle().getBlob();
           assert(blob && "Expecting dense resource with a valid blob");
-          rewriter.replaceOpWithNewOp<arith::ConstantOp>(
-              op, DenseElementsAttr::get(shapedType, blob->getData()));
+          auto elType = elements.getType().getElementType();
+          if (elType.isF32() || elType.isF16() || elType.isInteger(1) ||
+              elType.isInteger(8) || elType.isInteger(16) ||
+              elType.isInteger(32) || elType.isInteger(64)) {
```

Comment:
This is getting a bit hard to understand. What is the reason for not using `DenseResourceElementsAttr` always?

Reply:
Hey Ramiro, basically DenseResourceElementsAttr only supports certain data types. Specifically, it supports DenseBoolResourceElementsAttr, DenseI8ResourceElementsAttr, DenseI16ResourceElementsAttr, DenseI32ResourceElementsAttr, DenseI64ResourceElementsAttr, DenseUI8ResourceElementsAttr, DenseUI16ResourceElementsAttr, DenseUI32ResourceElementsAttr, DenseUI64ResourceElementsAttr, DenseF32ResourceElementsAttr, and DenseF64ResourceElementsAttr. So, when we run into cases with subbyte types from non trivial models, we would want to go through DenseElements. DenseElements supports everything, but we don't want to miss out on the optimization through DenseResource when we can.

Reply:
Is this documented somewhere? It would be nice to have a comment with a link to the supported Dense Elements Attrs. Also, how are you getting sub-byte types into torch-mlir? The standard PyTorch types are https://pytorch.org/docs/stable/tensor_attributes.html, which (excluding the boolean case), does not include any sub-byte integer types.

Reply:
Thanks for that clarification Ramiro. DenseResource can support the types we will be seeing in torch-mlir, and the code has been updated to remove the unnecessary checks.

Reply:
Great, thanks!